File: learn_c/17-07-25/reverse_digits.c
#include <stdio.h>

int main(void)
{
    int number, right_digit;

    printf("Enter your number:\n");
    scanf("%i", &number);

    if (number < 0)
    {
        printf("\n");      // Print minus sign
        number = -number; // Make number positive
    }

    do
    {
        right_digit = number % 10;
        printf("%i", right_digit);
        number = number / 10;
    } while (number != 0);

    printf("\n");
    return 0;
}

File: learn_python/30-07-25/reverse_digits.py
try:
    number = int(input("Enter your number: "))
except ValueError:
    print("Invalid number")
    exit()

sign = -1 if number < 0 else 1  # Save the sign
number = abs(number)           # Work with the positive part

reversed_number = 0

while number > 0:
    digit = number % 10
    reversed_number = reversed_number * 10 + digit
    number = number // 10

# Reapply the sign
reversed_number *= sign

print(reversed_number)

